src/decoder.c

```diff
@@ -239,6 +239,29 @@ int nanocbor_get_tag(nanocbor_value_t *cvalue, uint32_t *tag)
     return res;
 }
 
+int nanocbor_get_decimal_frac(nanocbor_value_t *cvalue, int32_t *e, int32_t *m)
+{
+    int res = NANOCBOR_NOT_FOUND;
+    uint32_t tag = UINT32_MAX;
+    if(nanocbor_get_tag(cvalue, &tag) == 0) {
+        if (tag == NANOCBOR_TAG_DEC_FRAC) {
+            nanocbor_value_t arr;
+            if (nanocbor_enter_array(cvalue, &arr) == 0) {
+                res = nanocbor_get_int32(&arr, e);
+                if (res >= 0) {
+                    res = nanocbor_get_int32(&arr, m);
+                    if (res >= 0) {
+                        res = NANOCBOR_OK;
+                    }
+                }
+            }
+            nanocbor_leave_container(cvalue, &arr);
```

Comment:
```suggestion
                nanocbor_leave_container(cvalue, &arr);
            }
```
This should be within the `nanocbor_enter_array` condition, otherwise the container can be leaved without entering it.

Reply:
Yep I had spotted it as well.